osu.Game/Screens/Multi/Multiplayer.cs

```diff
@@ -280,11 +280,14 @@ private void updateTrack(ValueChangedEvent<WorkingBeatmap> _ = null)
 
                 if (track != null)
                 {
-                    track.RestartPoint = Beatmap.Value.Metadata.PreviewTime;
                     track.Looping = true;
 
                     if (!track.IsRunning)
+                    {
+                        track.Stop();
+                        track.RestartPoint = Beatmap.Value.Metadata.PreviewTime != -1 ? Beatmap.Value.Metadata.PreviewTime : 0.4f * track.Length;
```

Comment:
Would it be worth to change the getter of `PreviewTime` to contain this logic instead of having it at each occurrence?